libcloud/dns/drivers/nsone.py

```diff
@@ -0,0 +1,328 @@
+        }
+        if extra is not None and extra.get('answers'):
+            raw_data['answers'] = extra.get('answers')
+        post_data = json.dumps(raw_data)
+        try:
+            response = self.connection.request(action=action, method='PUT',
+                                               data=post_data)
+        except NsOneException, e:
+            if e.message == 'record already exists':
+                raise RecordAlreadyExistsError(value=e.message, driver=self,
+                                               record_id='')
+            else:
+                raise e
+        record = self._to_record(item=response.parse_body(), zone=zone)
+
+        return record
+
+    def update_record(self, record, name, type, data, extra=None):
+        """
+        :param record: Record to update
+        :type record: :class:`Record`
+        :param name: Name of the record to update (e.g. foo).
+        :type name: ``str``
+        :param type: Type of record (e.g. A, MX etc)
+        :type type: ``str``
+        :param data: Data of the record (e.g. 127.0.0.1 for the A record)
+        :type data: ``str``
+        :param extra: Extra data needed to create different types of records
+        :type extra: ``dict``
+        :return: :class:`Record`
+        """
+        zone = record.zone
+        action = '/v1/zones/%s/%s/%s' % (zone.domain, '%s.%s' %
+                                         (name, zone.domain), type)
+        raw_data = {
+            "answers": [
+                {
+                    "answer": [
+                        data
+                    ], }
+            ]
+        }
+        if extra is not None and extra.get('answers'):
+            raw_data['answers'] = extra.get('answers')
+        post_data = json.dumps(raw_data)
+        try:
+            response = self.connection.request(action=action, data=post_data,
+                                               method='POST')
+        except NsOneException, e:
+            if e.message == 'record does not exist':
+                raise RecordDoesNotExistError(value=e.message, driver=self,
+                                              id=record.id)
+            else:
+                raise e
+        record = self._to_record(item=response.parse_body(), zone=zone)
+
+        return record
+
+    def ex_zone_exists(self, zone_id, zones_list):
+        """
+        Function to check if a `Zone` object exists.
+        :param zone_id: ID of the `Zone` object.
+        :type zone_id: ``str``
+
+        :param zones_list: A list containing `Zone` objects.
+        :type zones_list: ``list``.
+
+        :rtype: Returns `True` or `False`.
+        """
+        zone_ids = []
+        for zone in zones_list:
+            zone_ids.append(zone.id)
+
+        return zone_id in zone_ids
+
+    def _to_zone(self, item):
+        common_attr = ['zone', 'id', 'type']
+        extra = {}
+        for key in item:
+            if key not in common_attr:
```

Comment:
If a key is not in common_attr list then we add it in the extra dict as a key with the item.get('key') as value. The keys in common_attr are needed to build the Zone object. Why do you think we should drop it?